libraries/reconstruction_cuda/cuda_gpu_geo_transformer.cpp

```diff
@@ -349,20 +375,37 @@ void GeoTransformer<T>::checkRestrictions(int splineDegree,
         const std::pair<Matrix1D<T>, Matrix1D<T>> &coeffs, size_t frameIdx) {
     if (!isReadyForBspline)
         throw std::logic_error("Transformer is not ready yet.");
+
+    checkRestrictions(output, input);
+
+    if (frameIdx > inN)
+        throw std::invalid_argument("Frame index is out of bound");
+    size_t coeffsElems = splineX * splineY * splineN;
+    if ((coeffs.first.size() != coeffsElems) || (coeffs.second.size() != coeffsElems))
+        throw std::invalid_argument("Number of coefficients does not fit. "
+                "To init function, pass N control points.");
+}
+
+template<typename T>
+template<typename T_IN>
```

Comment:
I think we said that the input array type will have to match the transformer type, no?